### PR TITLE
Support slider step value extension.

### LINKED ISSUE #1260: Implement slider step value extension
**Current issue**
By default, the slider step increment is 10. It should be controllable from the questionnaire. This required implementation of an extension: http://build.fhir.org/extension-questionnaire-sliderstepvalue.html

**Describe the feature**
When using the slider, I should be able to control the steps increment

**Component**
Core library, SDC library, reference app, or SDC gallery app

**Expected behavior**
For example, when specifying a question like, how many previous pregnancies have you had, the slider should step up in increments of 1 (default is 10) - see screenshot of current behaviour below)

**Screenshots**
image

**Smartphone (please complete the following information):**
 - Device: [e.g. Pixel4a emulator]
 - Android version: [e.g. Settings -> About phone -> Android version]
 - Build number: [e.g. Settings -> About phone -> Build number]

**Additional context**

**Would you like to work on the issue?**
Assigning this bug to @santosh-pingle since he is already working on catalog app changes.

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1260

**Description**

- support slider step value extension in sdc module.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Feature

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt [hunk 1]
```diff
@@ -1495,6 +1496,25 @@ class MoreQuestionnaireItemComponentsTest {
       }
   }
 
+  @Test
+  fun `sliderStepValue is integer value if slider-step-value extension is present`() {
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt:1500]:
```suggestion
  fun `sliderStepValue should return the integer value in the sliderStepValue extension`() {
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt [hunk 2]
```diff
@@ -1495,6 +1496,25 @@ class MoreQuestionnaireItemComponentsTest {
...
+      Questionnaire.QuestionnaireItemComponent().apply {
+        linkId = "slider-step-value"
+        addExtension(EXTENSION_SLIDER_STEP_VALUE_URL, IntegerType(1))
+      }
+
+    assertThat(questionnaireItem.sliderStepValue).isEqualTo(1)
+  }
+
+  @Test
+  fun `sliderStepValue is null if slider-step-value extension is not present`() {
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt:1511]:
```suggestion
  fun `sliderStepValue should return null if sliderStepValue extension is not present`() {
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt [hunk 3]
```diff
@@ -74,6 +75,37 @@ class QuestionnaireItemSliderViewHolderFactoryTest {
     assertThat(viewHolder.itemView.findViewById<Slider>(R.id.slider).value).isEqualTo(10)
   }
 
+  @Test
+  fun `slider-step-value extension updates slider step size `() {
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt:79]:
```suggestion
  fun `step size should come from the sliderStepValue extension`() {
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt [hunk 4]
```diff
@@ -74,6 +75,37 @@ class QuestionnaireItemSliderViewHolderFactoryTest {
...
+        validationResult = NotValidated,
+        answersChangedCallback = { _, _, _ -> },
+      )
+    )
+
+    assertThat(viewHolder.itemView.findViewById<Slider>(R.id.slider).stepSize).isEqualTo(1)
+  }
+
+  @Test
+  fun `if slider-step-value extension is not present then default value is assigned as step size`() {
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt:96]:
```suggestion
  fun `step size should be 1 if the sliderStepValue extension is not present`() {
```

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactory.kt [hunk 5]
```diff
@@ -74,3 +78,5 @@ internal object QuestionnaireItemSliderViewHolderFactory :
       }
     }
 }
+
+private const val SLIDER_DEFAULT_STEP_SIZE = 10
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactory.kt:82]: let's make this 1. it's more usable by default.

```suggestion
private const val SLIDER_DEFAULT_STEP_SIZE = 1
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt [hunk 6]
```diff
@@ -1495,6 +1496,25 @@ class MoreQuestionnaireItemComponentsTest {
       }
   }
 
+  @Test
+  fun `sliderStepValue is integer value if slider-step-value extension is present`() {
+    val questionnaireItem =
+      Questionnaire.QuestionnaireItemComponent().apply {
+        linkId = "slider-step-value"
+        addExtension(EXTENSION_SLIDER_STEP_VALUE_URL, IntegerType(1))
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt:1504]:
```suggestion
        addExtension(EXTENSION_SLIDER_STEP_VALUE_URL, IntegerType(10))
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt [hunk 7]
```diff
@@ -1495,6 +1496,25 @@ class MoreQuestionnaireItemComponentsTest {
...
+      }
+
+    assertThat(questionnaireItem.sliderStepValue).isEqualTo(1)
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt:1507]:
```suggestion
    assertThat(questionnaireItem.sliderStepValue).isEqualTo(10)
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt [hunk 8]
```diff
@@ -74,6 +75,37 @@ class QuestionnaireItemSliderViewHolderFactoryTest {
     assertThat(viewHolder.itemView.findViewById<Slider>(R.id.slider).value).isEqualTo(10)
   }
 
+  @Test
+  fun `slider-step-value extension updates slider step size `() {
+    viewHolder.bind(
+      QuestionnaireItemViewItem(
+        Questionnaire.QuestionnaireItemComponent().apply {
+          linkId = "slider-step-value"
+          addExtension(EXTENSION_SLIDER_STEP_VALUE_URL, IntegerType(1))
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt:84]:
```suggestion
          addExtension(EXTENSION_SLIDER_STEP_VALUE_URL, IntegerType(10))
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt [hunk 9]
```diff
@@ -74,6 +75,37 @@ class QuestionnaireItemSliderViewHolderFactoryTest {
...
+        },
+        QuestionnaireResponse.QuestionnaireResponseItemComponent(),
+        validationResult = NotValidated,
+        answersChangedCallback = { _, _, _ -> },
+      )
+    )
+
+    assertThat(viewHolder.itemView.findViewById<Slider>(R.id.slider).stepSize).isEqualTo(1)
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt:92]:
```suggestion
    assertThat(viewHolder.itemView.findViewById<Slider>(R.id.slider).stepSize).isEqualTo(10)
```

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt [hunk 10]
```diff
@@ -74,6 +75,37 @@ class QuestionnaireItemSliderViewHolderFactoryTest {
...
+    viewHolder.bind(
+      QuestionnaireItemViewItem(
+        Questionnaire.QuestionnaireItemComponent().apply { linkId = "slider-step-value" },
+        QuestionnaireResponse.QuestionnaireResponseItemComponent(),
+        validationResult = NotValidated,
+        answersChangedCallback = { _, _, _ -> },
+      )
+    )
+
+    assertThat(viewHolder.itemView.findViewById<Slider>(R.id.slider).stepSize).isEqualTo(10)
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt:106]:
```suggestion
    assertThat(viewHolder.itemView.findViewById<Slider>(R.id.slider).stepSize).isEqualTo(1)
```

**Review by jingtang10 — APPROVED** [review]
thanks @santosh-pingle

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt (hunk 1 above)
Comment (santosh-pingle) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt]: done.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt (hunk 3 above)
Comment (santosh-pingle) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt]: done.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt (hunk 7 above)
Comment (santosh-pingle) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt]: done.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt (hunk 8 above)
Comment (santosh-pingle) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt]: done.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt (hunk 9 above)
Comment (santosh-pingle) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt]: done.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt (hunk 10 above)
Comment (santosh-pingle) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt]: done.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt (hunk 4 above)
Comment (santosh-pingle) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemSliderViewHolderFactoryTest.kt]: done.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt (hunk 6 above)
Comment (santosh-pingle) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt]: done.

File: datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt (hunk 2 above)
Comment (santosh-pingle) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/MoreQuestionnaireItemComponentsTest.kt]: done.

**Pull request merged**